packages/gatsby/src/db/sanitize-node.ts

```diff
@@ -20,15 +51,15 @@ const sanitizeNode = (data, isNode = true, path = new Set()) => {
         returnData[key] = o
         return
       }
-      returnData[key] = sanitizeNode(o, false, path)
+      returnData[key] = sanitizeNode(o as T, false, path)
```

Comment:
This may not be an `IGatsbyNode`, or even whatever `Node` is, as it's an arbitrary object inside a node so casting it is probably not what we want here?